### PR TITLE
Improvements to partial payments

### PR DESCRIPTION
**Pull request by stnatic**
I want to merge this change because they have been flagged in testing.

Changes that are introduced:
- `NOT_CHARGED` payments are deactivated on order creation
- validating amount of a non-partial payment does not take into account the amount covered by existing payments
- use PaymentErrors to transmit information about failed payments (SALEOR-4756)
- prevent calling `manager.process_payment` on an already completed payment

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4756 Refactor of the partial payments](https://app.clickup.com/t/2549495/SALEOR-4756)

File: saleor/checkout/complete_checkout.py [hunk 1]
```diff
@@ -406,8 +406,14 @@ def _create_order(
...
-    # assign checkout payments to the order
-    checkout.payments.update(order=order)
+    # assign checkout payments to the order and deactivate unused payments
+    payments = checkout.payments.all()
+    for payment in payments:
+        payment.order = order
+        if payment.charge_status == ChargeStatus.NOT_CHARGED:
+            payment.is_active = False
+
+    Payment.objects.bulk_update(payments, fields=["order", "is_active"])
```
Comment (korycins) [inline saleor/checkout/complete_checkout.py:416]: What about field `Paymeny.modified`?

File: saleor/graphql/checkout/tests/test_checkout_complete.py [hunk 2]
```diff
@@ -943,6 +943,53 @@ def test_checkout_complete_with_payment_id(
...
+    payment_dummy_fully_charged,
+):
+    # given
+    checkout = checkout_with_item
+    checkout.shipping_address = address
+    checkout.shipping_method = shipping_method
+    checkout.billing_address = address
+    checkout.save()
+
+    # manager = get_plugins_manager()
```
Comment (korycins) [inline saleor/graphql/checkout/tests/test_checkout_complete.py:960]: 🚔

**Review by korycins — APPROVED** [review]

File: saleor/checkout/tests/test_checkout_complete.py [hunk 3]
```diff
@@ -923,3 +924,42 @@ def test_create_order_use_translations(
...
+        **{
+            **payment_kwargs,
+            **{
+                "checkout": checkout_with_item,
+                "order": None,
+            },
+        }
+    )
+    payments_count = Payment.objects.count()
+    # when
```
Comment (IKarbowiak) [inline saleor/checkout/tests/test_checkout_complete.py:949]: I will suggest adding empty lines before `# when` and `# then` comments to separate different parts.

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**